src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -1486,6 +1495,136 @@ class spell_dru_wild_growth : public SpellScriptLoader
+                else if (spellInfo->Id == wraith)
+                    OnSpellCast(SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_LUNAR_AURA);
+            }
+        }
+    }
+
+    void HandleApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        SetSpellCount(SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, 2);
+        SetSpellCount(SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, 2);
+    }
+
+    void HandleRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* caster = GetCaster();
+        caster->RemoveAura(SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT);
+        caster->RemoveAura(SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT);
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_dru_eclipse_dummy::HandleApply, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        OnEffectRemove += AuraEffectApplyFn(spell_dru_eclipse_dummy::HandleRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        OnProc += AuraProcFn(spell_dru_eclipse_dummy::HandleProc);
+    }
+private:
+    void SetSpellCount(uint32 spellId, uint32 amount)
+    {
+        Unit* caster = GetCaster();
+        Aura* aura = caster->GetAura(spellId);
+        if (!aura)
+            caster->CastCustomSpell(spellId, SPELLVALUE_AURA_STACK, amount, nullptr, TRIGGERED_FULL_MASK);
+        else
+            aura->SetStackAmount(amount);
+    }
+
+    void OnSpellCast(uint32 cntSpellId, uint32 otherCntSpellId, uint32 eclipseAuraSpellId)
+    {
+        Unit* caster = GetCaster();
+        if (Aura* aura = caster->GetAura(cntSpellId))
+        {
+            uint32 remaining = aura->GetStackAmount();
+            if (remaining == 0)
+                return;
+
+            if (remaining > 1)
+                aura->SetStackAmount(remaining - 1);
+            else
+            {
+                // cast eclipse
+                caster->CastSpell(caster, eclipseAuraSpellId, TRIGGERED_FULL_MASK);
+
+                // Remove stacks from other one as well
+                // reset remaining power on other spellId
+                caster->RemoveAura(cntSpellId);
+                caster->RemoveAura(otherCntSpellId);
+            }
+        }
+    }
+};
+
+// 48517 Eclipse (Solar) + 48518 Eclipse (Lunar)
+class spell_dru_eclipse_occ : public AuraScript
+{
+    PrepareAuraScript(spell_dru_eclipse_occ);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT });
+    }
+
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+    }
+
+    void HandleRemoved(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (Player* player = GetCaster()->ToPlayer())
```

Comment:
Don't restrict things to Player only if you aren't using functions from Player class